Change E2E_MIN_STARTUP_PODS to 8 for GCE/GKE/AWS environments

girishkalele opened the pull request:
Test flake caused due to window of time between initial kube-system pods and the addon pods start running. Only 6 pods were found and they were all running. We need to wait for a minimum of 8 pods to be ready before we declare the cluster ready.

Fixes https://github.com/kubernetes/kubernetes/issues/26180

ixdy commented on jenkins/job-configs/global.yaml:
```diff
@@ -141,12 +141,13 @@
     gce-provider-env: |
         export KUBERNETES_PROVIDER="gce"
-        export E2E_MIN_STARTUP_PODS="1"
+        export E2E_MIN_STARTUP_PODS="8"
```
can you add a comment explaining where 8 comes from?

girishkalele commented:
@ixdy 

Updated with comment

goltermann commented:
This looks like the fix to a P0 (#/26180).  If so, can we get it merged ASAP?

girishkalele commented:
Copied P0 label over.

ixdy commented:
LGTM, but needs rebase.

girishkalele commented:
Ouch, rebase tricky because these *-provider-env blocks have been removed from global.yaml.

commit 24222a1873d515b692a87cde1f06266960c4f926
Author: Erick Fejta <fejta@google.com>
Date:   Wed Jun 8 14:44:36 2016 -0700

    Delete *-provider-env from global.yaml; migrate env into provider-env in e2e yamls

girishkalele commented:
@ixdy PTAL

Rebased and moved changes to the refactored code blocks.

wojtek-t commented on jenkins/job-configs/kubernetes-jenkins/kubernetes-kubemark.yaml:
```diff
@@ -7,7 +7,7 @@
     provider-env: |
         export KUBERNETES_PROVIDER="gce"
-        export E2E_MIN_STARTUP_PODS="1"
+        export E2E_MIN_STARTUP_PODS="8"
```
Im afraid that will not work, because we are not starting all addons in kubemark
@gmarek

gmarek replied:
I'll need to check, but I think we just ignore this variable.

girishkalele replied:
I can remove this change to the kubemark parameters.
Original change was in global.yaml but after rebase, I tried to add it to all provider environments.

ixdy replied:
let's leave this off of kubemark for now

ixdy commented on jenkins/job-configs/kubernetes-jenkins/kubernetes-soak.yaml:
```diff
@@ -163,7 +163,7 @@
             provider-env: |
                 export KUBERNETES_PROVIDER="gce"
-                export E2E_MIN_STARTUP_PODS="1"
+                export E2E_MIN_STARTUP_PODS="8"
```
should we expect 8 for 1.2 as well?

ixdy commented on jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-aws.yaml:
```diff
@@ -21,7 +21,7 @@
     provider-env: |
         export KUBERNETES_PROVIDER="aws"
-        export E2E_MIN_STARTUP_PODS="1"
+        export E2E_MIN_STARTUP_PODS="8"
```
it might be worth replicating the comment describing where this number came from.

long-term I wonder if we should have a smarter way to list the pods we expect so we don't have to try to keep a context-free number up-to-date.

girishkalele commented:
@ixdy 

Removed kubemark and 1.2 branch changes.

I agree, we need some way to figure out the expected number of pods in the kube-system namespace. Currently, there is a small window of time between the first 6 kube-system pods being ready and the addon manager loop detecting and launching the DNS and other static pods...if there was an externally observable event that the addon manager had run and added static pods on all nodes, we can make the e2e framework wait for that signal before checking all pods ready.

girishkalele commented:
@wojtek-t @ixdy @gmarek 

Updated...please take a look.

ixdy commented on jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-aws.yaml:
```diff
@@ -19,9 +19,10 @@
         - email-ext:
             recipients: '{emails}'
         - gcs-uploader
+    # Need a minimum of 8 kube-system pods ready before declaring cluster ready
```
sorry to continue to nit over comments, but this doesn't really explain where the value of 8 comes from.

girishkalele replied:
These are the set of essential kube-system pods necessary for E2E test suites to succeed. I will condense this list and reword all the comments.

```
etcd-server-kubernetes-master               
kube-apiserver-kubernetes-master            
kube-controller-manager-kubernetes-master   
kube-dns-v15-gandm                          
kube-scheduler-kubernetes-master            
l7-default-backend-v1.0-66izg               
l7-lb-controller-v0.6.3-kubernetes-master
kube-addon-manager-kubernetes-master        
```

ixdy replied:
yeah, basically I want to make sure that reading this config 3 months from now we understand why 8 was chosen as the magic value.

ixdy commented:
LGTM

fyi @spxtr @k8s-oncall
